jobs/slack-statistics.ts

```diff
@@ -18,16 +18,187 @@ export async function postStatisticsToSlack() {
+        where: {
+            createdAt: { gte: begin, lte: end },
+            knowsCoronaSchoolFrom: { notIn: [''] },
+        },
+        _count: {
+            createdAt: true,
+        },
+    });
+
+    const instructorKnowsCoronaSchoolFromValue = instructorKnowsCoronaSchoolFrom.map(function (item) {
+        return item['knowsCoronaSchoolFrom'];
+    });
+
+    const instructorKnowsCoronaSchoolFromCount = instructorKnowsCoronaSchoolFrom.map(function (item) {
+        return String(item['_count']['createdAt']);
+    });
+
+    const instructorKnowsCoronaSchoolFromTable: [string, string][] = [];
+    for (let i = 0; i < instructorKnowsCoronaSchoolFromValue.length; i++) {
+        instructorKnowsCoronaSchoolFromTable[i] = [instructorKnowsCoronaSchoolFromValue[i], instructorKnowsCoronaSchoolFromCount[i]];
+    }
+
+    // Matches
+    const matchCount = await prisma.match.count({
+        where: {
+            createdAt: { gte: begin, lte: end },
+        },
+    });
+
+    // New subcourses created
+    const subcourseCreatedCount = await prisma.subcourse.count({
+        where: {
+            createdAt: { gte: begin, lte: end },
+        },
+    });
+
+    // Number of current subcourses
+    const subcourseCurrentCount = await prisma.subcourse.count();
+
+    // TODO: number of free / taken seats of current public subcourses
+    const subcourseSeats = await prisma.course.findMany({
+        select: {
+            id: true,
+            name: true,
+            subcourse: {
+                select: {
+                    maxParticipants: true,
+                    // eslint-disable-next-line camelcase
+                    subcourse_participants_pupil: {
+                        select: {
+                            pupilId: true,
+                        },
+                    },
+                },
+            },
+        },
+    });
+
+    const subcourseSeatsCount = subcourseSeats.map((course) => {
+        const subcourses = course.subcourse;
+        const totalSeats = subcourses.reduce((acc, subcourse) => acc + subcourse.maxParticipants, 0);
+        const takenSeats = subcourses.reduce((acc, subcourse) => acc + subcourse.subcourse_participants_pupil.length, 0);
+        const availableSeats = totalSeats - takenSeats;
+
+        return {
+            courseName: course.name,
+            totalSeats,
+            takenSeats,
+            availableSeats,
+        };
+    });
+
+    const subcourseSeatsTakenTable: [string, string][] = [];
+    for (let i = 0; i < subcourseSeatsCount.length; i++) {
+        subcourseSeatsTakenTable[i] = [
+            subcourseSeatsCount[i].courseName,
+            String(subcourseSeatsCount[i].takenSeats + ' von ' + subcourseSeatsCount[i].totalSeats),
+        ];
+    }
+
+    // Match appointments
+    const matchAppointmentCount = await prisma.lecture.count({
+        where: {
+            createdAt: { gte: begin, lte: end },
+            matchId: { notIn: 0 },
```

Comment:
Does `notIn: 0` work? I think the matchId is `null` in case the appointment does not belong to a match. So I guess it's rather `{ matchId: { not: { equals: null } }`